documentation/wallet.adoc

```diff
@@ -0,0 +1,853 @@
+                return password
+            wallet_core -> wallet_core: Execute //Unseal Private Key//
+            wallet_core -> wallet_core: Execute //Generate message signature//
+            return message signature
+        return message signature
+    return message signature
+dapp -> dapp: Verify signature
+dapp -> dapp: Open user access
+return logout
+@enduml
+----
+
+===== Backup Account
+The following diagram illustrates the sequence of events when backing up an account.
+
+[plantuml, format="svg", id="backup-account", width="100%", role="center-image", options="interactive"]
+----
+@startuml
+
+skinparam FontName Poppins
+
+skinparam sequence {
+    ArrowColor #69679D
+    LifeLineBorderColor #69679D
+    LifeLineBackgroundColor #F5F5FF
+
+    ParticipantBorderColor #69679D
+    ParticipantBackgroundColor #F5F5FF
+    ParticipantFontName Poppins
+    ParticipantFontSize 16
+    ParticipantFontColor #000000
+}
+
+participant "User" as user
+participant "Massa Wallet Front" as wallet_front
+participant "Massa Wallet API" as wallet_api
+participant "Massa Wallet Core" as wallet_core
+participant "Massa Wallet Embedded Front" as wallet_embedded_front
+
+user -> wallet_front ++ : Click "Backup Account"
+    wallet_front -> wallet_api ++: Request //Backup Account//
+        wallet_api -> wallet_core ++: Execute //Backup Account//
+            wallet_core -> wallet_embedded_front ++: Fetch //Password Input//
+            wallet_embedded_front <-> user: Enter password
+            return password
+        wallet_core -> wallet_core: Execute //Unseal Private Key//
+        wallet_core -> wallet_embedded_front: Execute //Display Private Key//
+        wallet_embedded_front <-> user: Read/Copy Private Key
+        user -> wallet_embedded_front: Close
+        return Success
+    return Success
+return Success
+@enduml
+----
+
+<<<
+
+===== Add asset
+The following diagram illustrates the sequence of events when adding an asset.
+
+[plantuml, format="svg", id="add-asset", width="100%", role="center-image", options="interactive"]
+----
+@startuml
+
+skinparam FontName Poppins
+
+skinparam sequence {
+    ArrowColor #69679D
+    LifeLineBorderColor #69679D
+    LifeLineBackgroundColor #F5F5FF
+
+    ParticipantBorderColor #69679D
+    ParticipantBackgroundColor #F5F5FF
+    ParticipantFontName Poppins
+    ParticipantFontSize 16
+    ParticipantFontColor #000000
+}
+
+participant "User" as user
+participant "Massa Wallet Front" as wallet_front
+participant "Massa Wallet API" as wallet_api
+participant "Massa Wallet Core" as wallet_core
+
+user -> wallet_front ++ : Click "Import Token"
+user -> wallet_front : Enter "Token Address"
+    wallet_front -> wallet_api ++: Request //Add Asset//
+        wallet_api -> wallet_core ++: Execute //Add Asset//
+            wallet_core -> wallet_core: Execute //Update Assets File//
+        return Asset
+    return Asset
+return Success
+@enduml
+----
+
+===== Create New Account
```

Comment:
i'm stopping seuqence diagramreview here. Please have a look at what we did on the lazy manager project to align on the practices.